src/sys/unix/kqueue.rs

```diff
@@ -306,7 +306,7 @@ impl Events {
                 }
             }
 
-            if e.flags & libc::EV_EOF != 0 {
+            if e.filter == libc::EVFILT_WRITE as Filter && e.flags & libc::EV_EOF != 0 {
```

Comment:
IMHO this conditional is now complex enough that it would be nice to have a comment explaining it?

Reply:
Restructured + added a comment.